### DISCUSSION TITLE
How to specify an overlay for pkgs ?

### DISCUSSION THREAD

**Discussion by gaelreyrol** [nix-community/nixvim#529] [Q&A] [answered]
Hi!

I am using multiple overlays over the standard pkgs attribute to access unstable & master channels:

```nix
nixpkgs.overlays = [
  (final: prev: {
    unstable = builtins.import inputs.unstable {
      inherit (final) config;
      inherit system;
    };
  })
  (final: prev: {
    master = builtins.import inputs.master {
      inherit (final) config;
      inherit system;
    };
  })
]
```

As I am using the master branch of nixvim, it expects pkgs to be the unstable channel. Mine targets nixos-23.05, I would like to know if there was a way to pass to nixvim.nixosModules.nixvim a specific pkgs attribute. For example:

```nix
modules = [
  nixvim.nixosModules.nixvim
  {
    nixvim.pkgs = pkgs.unstable;
  }
];
```

Thanks for your time!

**Comment by GaetanLepage** [comment]
Hello !
Actually, it depends on how you use nixvim.
If you use it as a standalone flake, then you can provide it with any `nixpkgs` of your choice.
However, if you use the nixvim home-manager or NixOS module, then it will inherit whatever is your system `pkgs`.

In the end, if your system `pkgs` is `nixos-23.05`, I am afraid that you are constrained to using the `nixos-23.05` branch of nixvim.

**Comment by GaetanLepage** [comment, marked as answer]
We currently don't backport anything but bugfixes to the stable branch. Hence, the `nixos-23.05` branch of nixvim is more or less untouched since when it has been created (in may).
If you really need newer plugins and additions to nixvim, then either you switch your system `nixpkgs` to an unstable channel, or you use `nixvim` as a standalone flake (see [here](https://github.com/nix-community/nixvim#usage)).

**Comment by gaelreyrol** [comment]
Thanks for your answers, I managed to use the standalone method instead of the module to provide my own channel !
